### PR TITLE
Update scaffolding template procedure

### LINKED ISSUE #122: Update fetching scaffold example from client directly instead of calling controller
Currently in `skycli app scaffold` command, skycli call controller to get the example list and download the selected example from controller. Controller is calling github api to get the example, but there are rate limit for github api. The limit is 60 request per hour for unauthenticated requests. To keep it simple, maybe we should not make the call in controller, just call the github api directly in client.

### LINKED ISSUE #153: Use OS-specific config directory path
Current we defined config directory to be `${XDG_CONFIG_HOME:-$HOME/.config}` which is not expected in Window. To get OS-specific config path, we can use this lib: https://github.com/sindresorhus/env-paths

### PR DESCRIPTION
**Pull request by kiootic**
fix #122 
fix #153

### CODE REVIEW

File: src/commands/app/scaffold.ts [hunk 1]
```diff
@@ -30,23 +37,39 @@ async function selectApp(argv: Arguments): Promise<string> {
...
+  let localVersion = currentVersion;
+
+  if (latestVersion && currentVersion !== latestVersion) {
+    console.log('Updating templates...');
+    try {
+      await updateTemplates(latestVersion);
+    } catch (error) {
+      console.log(chalk`{yellow WARN:} Failed to update templates`);
+    }
+    localVersion = latestVersion;
```
Comment (carmenlau) [inline src/commands/app/scaffold.ts]: This line should be inside `try`?

File: package.json [hunk 2]
```diff
@@ -35,10 +35,11 @@
   "homepage": "https://skygear.io/",
   "dependencies": {
     "@babel/polyfill": "^7.4.4",
-    "@skygear/node-client": "^2.0.0-alpha.11",
+    "@skygear/node-client": "^2.0.0-alpha.13",
     "@zeit/dockerignore": "0.0.5",
     "chalk": "^2.4.2",
     "cli-table3": "^0.5.1",
+    "env-paths": "2.2.0",
```
Comment (carmenlau) [inline package.json]: I found that `env-paths` changed the config path of mac to ` /Users/{user}/Library/Preferences/`.
I think it make sense for normal app, but not for command line program.
It is better if we can its linux version path in mac too, see if we want to implement ourselves?

Comment (kiootic) [reply package.json]: I think it make sense? We can also say the same thing for Windows platform, but I think following platform's convention on config location is better.

Comment (louischan-oursky) [reply package.json]: The conventional configuration location of command line program is universal for UNIX-like system.

**Review by carmenlau — APPROVED** [review]

**Pull request merged**